docs/api/naj-cookbook.md

```diff
@@ -0,0 +1,507 @@
+  nodeUrl: "https://rpc.testnet.near.org",
+  walletUrl: "https://wallet.testnet.near.org",
+  helperUrl: "https://helper.testnet.near.org",
+};
+
+const credentialsPath = path.join(homedir, CREDENTIALS_DIR);
+const keyStore = new nearAPI.keyStores.UnencryptedFileSystemKeyStore(
+  credentialsPath
+);
+
+sendTxs();
+
+async function sendTxs() {
+  const near = await nearAPI.connect({ ...config, keyStore });
+  const account = await near.account("example.testnet");
+  const newArgs = { staking_pool_whitelist_account_id: WHITELIST_ACCOUNT_ID };
+  const result = await account.signAndSendTransaction({
+    receiverId: CONTRACT_NAME,
+    actions: [
+      nearAPI.transactions.createAccount(),
+      nearAPI.transactions.transfer(TRANSFER_AMOUNT),
+      nearAPI.transactions.deployContract(fs.readFileSync(WASM_PATH)),
+      nearAPI.transactions.functionCall(
+        "new",
+        Buffer.from(JSON.stringify(newArgs)),
+        10000000000000,
+        "0"
+      ),
+    ],
+  });
+
+  console.log(result);
+}
+```
+
+---
+
+## Utils
+
+### Calculate Gas
+
+> Calculate the gas and tokens burnt from any contract call by looping through the `result` receipts.
+
+```js
+const { connect, KeyPair, keyStores, utils } = require("near-api-js");
+const path = require("path");
+const homedir = require("os").homedir();
+const chalk = require("chalk");
+
+const CREDENTIALS_DIR = ".near-credentials";
+const ACCOUNT_ID = "near-example.testnet";
+const CONTRACT_ID = "guest-book.testnet";
+const METHOD_NAME = "addMessage";
+const args = {
+  text: "Howdy!",
+};
+
+const config = {
+  networkId: "testnet",
+  nodeUrl: "https://rpc.testnet.near.org",
+  walletUrl: "https://wallet.testnet.near.org",
+  helperUrl: "https://helper.testnet.near.org",
+};
+
+calculateGas(CONTRACT_ID, METHOD_NAME, args, "0");
+
+async function calculateGas(contractId, methodName, args, depositAmount) {
+  const credentialsPath = path.join(homedir, CREDENTIALS_DIR);
+  const keyStore = new keyStores.UnencryptedFileSystemKeyStore(credentialsPath);
+  const near = await connect({ ...config, keyStore });
+  const account = await near.account(ACCOUNT_ID);
+  let gasBurnt = [];
+  let tokensBurnt = [];
+  const result = await account.functionCall({
+    contractId,
+    methodName,
+    args,
+    gas: "300000000000000",
+    attachedDeposit: utils.format.parseNearAmount(depositAmount),
+  });
+  gasBurnt.push(result.transaction_outcome.outcome.gas_burnt);
+  tokensBurnt.push(
+    utils.format.formatNearAmount(
+      result.transaction_outcome.outcome.tokens_burnt
+    )
+  );
+  for (let i = 0; i < result.receipts_outcome.length; i++) {
+    gasBurnt.push(result.receipts_outcome[i].outcome.gas_burnt);
+    tokensBurnt.push(
+      utils.format.formatNearAmount(
+        result.receipts_outcome[i].outcome.tokens_burnt
+      )
+    );
+  }
+  const totalGasBurned = gasBurnt.reduce((acc, cur) => acc + cur, 0);
+  const totalTokensBurned = tokensBurnt.reduce((acc, cur) => acc + cur, 0);  
```

Comment:
```suggestion
  const account = await near.account(ACCOUNT_ID);
  const result = await account.functionCall({
    contractId,
    methodName,
    args,
    gas: "300000000000000",
    attachedDeposit: utils.format.parseNearAmount(depositAmount),
  });

  const { totalGasBurned, totalGasBurned } = result.receipts_outcome.reduce(
    (acc, receipt) => {
      acc.totalGasBurned += receipt.outcome.gas_burnt;
      acc.totalTokensBurned += utils.format.formatNearAmount(receipt.outcome.tokens_burnt)
    },
    {
      totalGasBurned: result.transaction_outcome.outcome.gas_burnt,
      totalTokensBurned: utils.format.formatNearAmount(
        result.transaction_outcome.outcome.tokens_burnt
      ),
    }
  );

```

Reply:
Awesome! Nice refactor! I think on line 447 you meant `totalGasBurned, totalTokensBurned`. Updating this now